tools/isobuild/unibuild.ts

```diff
@@ -1,21 +1,77 @@
 "use strict";
 
 import _ from "underscore";
-import files from "../fs/files";
+import * as files from "../fs/files";
 import { WatchSet, sha1 } from "../fs/watch";
 import { NodeModulesDirectory } from "./bundler.js";
 import * as archinfo from "../utils/archinfo.js";
+import Isopack from "./isopack";
+import Builder from "./builder";
 
-function rejectBadPath(p) {
+function rejectBadPath(p: string) {
   if (p.indexOf("..") >= 0) {
     throw new Error("bad path: " + p);
   }
 }
 
 let nextBuildId = 1;
 
+type IsopackConstraint = {
+  package: string;
+  constraint: string;
+  unordered: boolean;
+  weak: boolean;
+};
+
+type PrelinkResource = {
+  type: string;
```

Comment:
This `PrelinkResource` type looks very similar to [the `File` type in `import-scanner.ts`](https://github.com/meteor/meteor/blob/f8ba7a96fe98b441a0f043f2ac5c7e98f56e7c3d/tools/isobuild/import-scanner.ts#L290-L308). I wonder if there's an opportunity to consolidate them?